src/Control/Email/Email.php

```diff
@@ -769,237 +409,132 @@ public function getHTMLTemplate()
-     * @return $this
-     */
-    public function setFailedRecipients($recipients)
-    {
-        Deprecation::notice('4.12.0', 'Will be removed without equivalent functionality to replace it');
-        $this->failedRecipients = $recipients;
-
         return $this;
     }
 
-    /**
-     * @deprecated 4.12.0 Will be removed without equivalent functionality to replace it
-     *
-     * @return array
-     */
-    public function getFailedRecipients()
-    {
-        Deprecation::notice('4.12.0', 'Will be removed without equivalent functionality to replace it');
-        return $this->failedRecipients;
-    }
-
-    /**
-     * @deprecated 4.12.0 Will be replaced with getData()
-     *
-     * Used by {@link SSViewer} templates to detect if we're rendering an email template rather than a page template
-     *
-     * @return bool
-     */
-    public function IsEmail()
-    {
-        Deprecation::notice('4.12.0', 'Will be replaced with getData()');
-        return true;
-    }
-
     /**
      * Send the message to the recipients
-     *
-     * @return bool true if successful or array of failed recipients
      */
-    public function send()
+    public function send(): void
     {
-        if (!$this->getBody()) {
-            $this->render();
-        }
-        if (!$this->hasPlainPart()) {
-            $this->generatePlainPartFromBody();
-        }
-        return Injector::inst()->get(Mailer::class)->send($this);
+        $this->updateHtmlAndTextWithRenderedTemplates();
+        Injector::inst()->get(MailerInterface::class)->send($this);
     }
 
     /**
-     * @return array|bool
+     * Send the message to the recipients as plain-only
      */
-    public function sendPlain()
+    public function sendPlain(): void
     {
-        if (!$this->hasPlainPart()) {
-            $this->render(true);
-        }
-        return Injector::inst()->get(Mailer::class)->send($this);
+        $html = $this->getHtmlBody();
+        $this->html(null);
+        $this->updateHtmlAndTextWithRenderedTemplates(true, $html);
+        Injector::inst()->get(MailerInterface::class)->send($this);
+        $this->html($html);
     }
 
     /**
-     * @deprecated 4.12.0 Will be removed without equivalent functionality to replace it
+     * Call html() and/or text() after rendering email templates
+     * If either body html or text were previously explicitly set, those values will not be overwritten
      *
-     * Render the email
-     * @param bool $plainOnly Only render the message as plain text
-     * @return $this
+     * @param bool $plainOnly - if true then do not call html()
+     * @param ?string $htmlFallback - HTML used to render plain-only email content when text() is not set.
+     *                                HTML tags will be removed from render.
      */
-    public function render($plainOnly = false)
+    private function updateHtmlAndTextWithRenderedTemplates(bool $plainOnly = false, ?string $htmlFallback = null): void
```

Comment:
Somewhere in this method (probably right at the end) should we be setting `dataHasBeenSet` to false again?
i.e. is it intended as a "changes were made since last time we rendered" to replicate the `invalidateBody()` call we used to have (https://github.com/silverstripe/silverstripe-framework/blob/4/src/Control/Email/Email.php#L645) or is it just a "there is some data beyond just the defaults" flag?